okdownload/src/main/java/com/liulishuo/okdownload/core/interceptor/RetryInterceptor.java

```diff
@@ -45,6 +45,9 @@
                 }
 
                 chain.getCache().catchException(e);
+                if (e != InterruptException.SIGNAL) {
```

Comment:
It's no need to except `InterruptException.SIGNAL`, because it will let `catchBlockConnectException` method decoupling with the case of `InterruptException`, and on the logic design, it's fine to catch all `IOException` on the internal of `catchBlockConnectException`

Reply:
But `catchBlockConnectException ` will be invoked multi times if don't except InterruptException.SIGNAL, that will cause the size of noMoreStreamList larger than needFetchBlockCount

Reply:
Or, I can handle duplicate logic in `catchBlockConnectException` method so that decoupling with the case of `InterruptException `.

Reply:
updated

Reply:
1. Please don't trust the count of `noMoreStreamList`, it will contain same block-index duplicate, and it is accepted. the `noMoreStreamList` is just can tell which one is `no-more-stream`.
2. If you rely on the `catchBlockConnectException` can be only invoked once, it's very bad design
3. For the current block-thread, I don't think this will be invoked more than once